c10/core/TensorImpl.h

```diff
@@ -563,6 +563,8 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   template <typename T>
   inline T * data() const {
     AT_ASSERT(!is_variable());  // TODO: remove this when Variable and Tensor are merged
+    AT_CHECK(has_storage(),
```

Comment:
did you benchmark perf of this? this is a virtual call currently so might have some overhead. Why not check that storage_ is not null? (which is what (bool)storage_ would do)

Reply:
No, I did not.